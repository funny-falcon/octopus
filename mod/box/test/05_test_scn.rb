#!/usr/bin/ruby1.9.1

$:.push 'test/lib'
require 'standalone_env'

class Env < StandAloneEnv
  def config
    super + <<EOD
rows_per_wal=10

object_space[0].enabled = 1
object_space[0].index[0].type = "HASH"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"
EOD
  end
end

env = Env.clean
env.with_server do
  ping
  30.times do insert [1] end

  puts File.open("00000000000000000010.xlog").lines.take_while {|l| l != "\n" } + ["\n"]
  puts `octopus --cat 00000000000000000010.xlog | sed 's/tm:[^ ]* //'`
  puts
  puts File.open("00000000000000000020.xlog").lines.take_while {|l| l != "\n" } + ["\n"]
  puts `octopus --cat 00000000000000000020.xlog | sed 's/tm:[^ ]* //'`
end
