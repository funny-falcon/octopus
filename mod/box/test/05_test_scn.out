# box.ping()
:pong

# box.insert([1])
1

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

# box.insert([1])
2

XLOG
0.12
Created-by: octopus
SCN: 10

lsn:10 scn:10 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:11 scn:11 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:12 scn:12 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:13 scn:13 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:14 scn:14 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:15 scn:15 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:16 scn:16 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:17 scn:17 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:18 scn:18 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:19 scn:19 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">

XLOG
0.12
Created-by: octopus
SCN: 20

lsn:20 scn:20 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:21 scn:21 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:22 scn:22 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:23 scn:23 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:24 scn:24 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:25 scn:25 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:26 scn:26 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:27 scn:27 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:28 scn:28 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
lsn:29 scn:29 t:wal/usr13 0.0.0.0:0 INSERT n:0 <1:"\x01\x00\x00\x00">
